pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -844,6 +852,18 @@ func (npm *nominatedPodMap) UpdateNominatedPod(oldPod *v1.Pod, newPodInfo *frame
 // NewPodNominator creates a nominatedPodMap as a backing of framework.PodNominator.
```

Comment:
We should mark this as deprecated, and eventually make all test use the real nominator.

After this change, which kind of tests still use this?

Reply:
Making this as deprecated sounds good.

Here are the tests using it:

```
pkg/scheduler/scheduler_test.go
834:		frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),

pkg/scheduler/core/extender_test.go
274:				runtime.WithPodNominator(internalqueue.NewPodNominator()),

pkg/scheduler/core/generic_scheduler_test.go
1000:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1063:		frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1096:		frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1170:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1332:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1435:		frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1521:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),

pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go
289:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1006:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1233:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
1638:				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),

pkg/scheduler/framework/runtime/framework_test.go
1482:			podNominator := internalqueue.NewPodNominator()
```

Reply:
why not just remove it in this PR?

Reply:
I'd leave it for a 2nd PR just so that this PR is easier to port

Reply:
sg if we are planning to cherrypick.